streams/src/main/java/org/apache/kafka/streams/state/internals/OffsetCheckpoint.java

```diff
@@ -76,7 +80,23 @@ public void write(final Map<TopicPartition, Long> offsets) throws IOException {
             // write to temp file and then swap with the existing file
             final File temp = new File(file.getAbsolutePath() + ".tmp");
 
-            final FileOutputStream fileOutputStream = new FileOutputStream(temp);
+            FileOutputStream fileOutputStream;
+
+            try {
+                fileOutputStream = new FileOutputStream(temp);
+            } catch (final FileNotFoundException e) {
+                LOG.warn("Checkpoint file doesn't exist will attempt to create [{}]", file, e);
+                final boolean createdParentDirs = temp.getParentFile().mkdirs();
+                final boolean createdNewFile = temp.createNewFile();
+
```

Comment:
Do we need to lock this directory to avoid further removal of the state directory?